### PR TITLE
move ZModulo to test suite

### PR DESCRIPTION
**Pull request by andres-erbsen**
https://github.com/coq/coq/pull/16914

- [x] Added / updated **test-suite**.
- [x] Added **changelog**.

### CODE REVIEW

**Comment by andres-erbsen** [comment]
@coqbot run full ci

**Comment by andres-erbsen** [comment]
elpi failure is also on master https://github.com/coq/coq/commit/8e0314415cc3cd528fc81bf11075d294179f880a

**Comment by proux01** [comment]
@andres-erbsen note that the usual policy before removing something is that potential users should have been able to see a deprecation warning for at least two versions. So I'm afraid we cannot remove that now.

**Comment by andres-erbsen** [comment]
Are you sure this is the policy (even for changes with backwards-compatible fixups)? The most relevant written record I can find says

>  it should always be possible for a project written in Coq to be compatible with two successive major versions, features must be deprecated in one major version before they can be removed in the next, Coq developers must have a good estimate of the effort it takes to fix a project with respect to a given change, and finally and most importantly, breaking changes must be clearly documented with clear recommendations explaining how to fix a project.

(from the dissertation of @Zimmi48, https://hal.inria.fr/tel-02451322v1 page 44)

Removals based on similar reasoning have been approved and merged as well, e.g. https://github.com/coq/coq/pull/15268.

**Comment by Alizter** [comment]
The fact that we cannot deprecate entire modules is a missing feature that would be very useful here.

**Comment by proux01** [comment]
@andres-erbsen indeed I was wrong with the requirement for deprecation in two releases, one can be enough under some circumstances.
I still think that we should offer users a deprecation warning in at least one release.

**Comment by Alizter** [comment]
Here is an issue about deprecating libraries https://github.com/coq/coq/issues/8032

**Comment by andres-erbsen** [comment]
The deprecation was announced in the previous release [here](https://github.com/coq/coq/pull/16914/files#diff-01103d30e1254bae3fc907e16ea09c9162cbe8111e2f5e2ad6393dd69573bc1dR1-R7). For a library that was actually used or understood to be useful, I can see that aiming for a higher level of communication (some solution to https://github.com/coq/coq/issues/8032, perhaps manual) would have value, but AFAIK this file has only caused confusion so far so I don't think it is worth holding back with its removal any longer.

**Comment by Zimmi48** [comment]
Given that CI reveals no issues, I also support this removal (but it should be documented in a changelog).

File: doc/changelog/11-standard-library/17258-remove-Zmodulo.rst
Comment (jfehrle) [inline doc/changelog/11-standard-library/17258-remove-Zmodulo.rst:3]:
```suggestion
- **Removed:** :g:`Numbers.Cyclic.ZModulo` from the standard library. This file
  was deprecated in 8.17 and has no known use cases. It is retained in
  the test suite to ensure consistency of :g:`CyclicAxioms`
```

Please at least correct the typo "ensurce".

Comment (andres-erbsen) [reply doc/changelog/11-standard-library/17258-remove-Zmodulo.rst]: Thank you!

**Comment by ppedrot** [comment]
AFAICT this is ready and needs an assignee. @proux01 maybe?

**Comment by proux01** [comment]
Maybe I should clarify my position here: I still think this deserves proper deprecation warnings and I won't merge without them (I would otheriwse) but I will not oppose if anyone else think it's fine without warnings.

**Comment by ppedrot** [comment]
Either we merge this quickly, or we do deprecate the relevant definitions before the 8.18 branch. Since there is not much momentum for the first option, maybe the second should be considered?

**Comment by JasonGross** [comment]
Now that 8.18 has branched, this can be merged into master, right?

**Comment by andres-erbsen** [comment]
Gentle ping @coq/number-maintainers

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot run full ci

**Comment by Alizter** [comment]
@voqbot run full ci

**Comment by andres-erbsen** [comment]
Full CI failures look unrelated, refman failure is fixed in latest version.

**Comment by Alizter** [comment]
@coqbot merge now

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**
